Feature common

asqwklop12 opened the pull request:
## PR 체크리스트
-  커밋 컨벤션에 맞게 작성했는가?
-  PR 전에 현재 브랜치를 pull 받았는가?
-  구현한 기술들을 문서화 하였는가?


<br>

## PR 작업 분류

<!-- Please check the one that applies to this PR using "x". -->

- [o] 신규 기능
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 기타


<br>

## 작업 상세 내용
- 공통 모듈을 개발하였습니다.
사용법 : 1. settings.gradle에 include 'project'를 추가한다.
             2. 오른쪽 그래들에서 해당 프로젝트를 삭제하고 실행한다.
             3. 해당 모듈 build.gralde에서 implementation project(':common')으로 추가해서 공통 모듈을 가져온다.
             4. 현재 공통 모듈에는 baseEntity만 존재하니 사용한다. 
             -끝-

<br>

## 🔴 이건 반드시 확인해 주세요!

asqwklop12 committed:
[feat] 공통 모듈 개발 - client,eureka 등록

asqwklop12 committed:
[build] 공통 모듈 product 등록

luz315 commented on common/src/main/java/com/sparta/rooibos/common/entity/BaseEntity.java:
```diff
@@ -0,0 +1,39 @@
+@Getter
+@Setter
+public abstract class BaseEntity {
+    @CreationTimestamp
```
Spring Data JPA의 Auditing 기능으로 @CreatedDate, @LastModifiedDate을 사용하시면 Spring에서 자동 처리되므로 주로 권장되는 방법입니다
그럼에도 불구하고 Hibernate 전용 기능인 @CreationTimestamp, @UpdateTimestamp를 사용하신 특별한 이유가 있을까요?

asqwklop12 replied:
아하 저건 그냥 1차때도 저렇게 사용해서 그렇게 사용하였습니다.

asqwklop12 replied:
좋은 정보 감사합니다.

luz315 commented on common/src/main/java/com/sparta/rooibos/common/entity/BaseEntity.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Column
+    private String deleteBy;
+}
```
```
  public void delete(String deletedBy) {
        this.deletedAt = LocalDateTime.now();
        this.deletedBy = deletedBy;      
        this.hidden = true;   
    }
```
해당 메서드를 추가해주신다면 더욱 간편히 사용할 수 있을 것 같습니다

asqwklop12 replied:
애초에 이거 같은 경우, 각 도메인에서 baseEntity를 사용할때, 문제 없이.. 사용이 되어진다면, 추가할 예정이었으나... 
각 도메인에서 baseEntity를 사용할때 Q객체를 만들지못해 일단은 그냥 뒀었는데..  (그전에 까먹고 그냥 올리긴 한것도 맞긴해서...) 깜빡하고 추가하지 못한거 같습니다.ㅜㅜㅜ 얘는 일단 반영해놓겠습니다.!!!

luz315 committed:
[feat] 롤체킹 aop 구현

luz315 committed:
[feat] 롤체킹 aop 구현

luz315 committed:
[feat] 롤체킹 aop 구현

asqwklop12 committed:
[refactor] 사용 어노테이션 변경
@CreationTimestamp -> @CreatedDate
@UpdateTimestamp -> @LastModifiedDate
스프링에서 제공하는걸로 변경
isDelete 추가

ekdh0858 reviewed (approved):
👍
